doc/acl/Dynamic-ACL-Design.md

```diff
@@ -0,0 +1,198 @@
+The new table contains all fileds of ACL rules and new fields are added: `is_dynamic`, `ttl`, `creation_time` and `expiration_time`.
+```
+key: ACL_RULE_TABLE:table_name:seq  ; key of the rule entry in the table, seq is the order of the rules   
+                                    ; when the packet is filtered by the ACL "policy_name".   
+                                    ; A rule is always assocaited with a policy.
+
+;field          = value
+is_dynamic      = true/false  ; Boolean; Optional. Indicate the rule is a time-based ACL rule.
+creation_time   = Integer     ; timestamp when the rule is created or refreshed
+expiration_time = Integer     ; timestamp when the rule is expired
+ttl             = Integer     ; the ttl, in second
+```
+A sample config for ACL rule
+```json
+{
+    "DYNAMIC_ACL_RULE_TABLE|DATAACL|DYNAMIC_RULE_1":{
+        "DST_IP":"192.168.0.3/32",
+        "ETHER_TYPE":"2048",
+        "PACKET_ACTION":"FORWARD",
+        "PRIORITY":"9999",
+        "SRC_IP":"192.168.0.2/32",
+        "creation_time": "1662432143",
+        "expiration_time": "1662432153",
+        "is_dynamic": "True",
+        "ttl": "10"
+    }
+}
+```
+A sample json config for ACL rule
+```json
+{
+    "acl":{
+        "acl-sets":{
+            "acl-set":{
+                "DATAACL":{
+                    "acl-entries":{
+                        "acl-entry":{
+                            "1":{
+                                "actions":{
+                                    "config":{
+                                        "forwarding-action":"ACCEPT"
+                                    }
+                                },
+                                "config":{
+                                    "sequence-id":1
+                                },
+                                "ip":{
+                                    "config":{
+                                        "source-ip-address":"192.168.0.2/32",
+                                        "destination-ip-address":"192.168.0.3/32"
+                                    }
+                                },
+                                "dynamic-acl":{
+                                    "config":{
+                                        "ttl":"10"
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+```
+The YANG of ACL_RULE is required to be updated to accept new field `is_dynamic`, `ttl`, `creation_time` and `expiration_time`
+
+Orchagent (actually `aclorch`) won't consume the value of the new flags. So no change is required to `orchagent`.
+#### APP DB
+Existing table `ACL_RULE_TABLE` is used to activate an effective ACL rule. The ACL rule format is the same as in CONFIG DB.
+
+A sample
+```json
+{
+    "APP_ACL_RULE|DATAACL|DYNAMIC_RULE_1":{
```

Comment:
Replace `DYNAMIC_RULE` to `TIME_RANGE_RULE`

Reply:
Accepted and thanks!